src/main/java/strategy/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,20 @@
+package strategy.operation;
+
+import dao.FruitShopDaoImpl;
+import exception.FruitShopException;
+import model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final FruitShopDaoImpl fruitShopDao = new FruitShopDaoImpl();
+
+    @Override
+    public void execute(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int balance = fruitShopDao.getBalanceByFruit(fruit) - transaction.getQuantity();
+        if (balance > 0) {
+            fruitShopDao.putBalanceStatistic(fruit, balance);
+        } else {
+            throw new FruitShopException("Quantity can't be bigger then balance");
+        }
```

Comment:
```suggestion
        if (balance <= 0) {
            throw new FruitShopException("Not enough in storage: " + balance);
        }
        fruitShopDao.putBalanceStatistic(fruit, balance);
```
throw/return early, it's a better pattern, you won't need `else` - indented code is harder to read [link](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement)